src/util/subprocess.h

```diff
@@ -309,25 +309,36 @@ namespace util
    * Parameters:
    * [in] str : Input string which needs to be split based upon the
    *            delimiters provided.
-   * [in] deleims : Delimiter characters based upon which the string needs
-   *                to be split. Default constructed to ' '(space) and '\t'(tab)
-   * [out] vector<string> : Vector of strings split at deleimiter.
+   * [in] delims : Delimiter characters based upon which the string needs
+   *               to be split. Default constructed to ' '(space) and '\t'(tab)
+   * [in] quote : Quote char that encloses a token. Default constructed
+   *              to "'" (single quote)
+   * [out] vector<string> : Vector of tokens split at delimiters and quotes.
    */
   static inline std::vector<std::string>
-  split(const std::string& str, const std::string& delims=" \t")
+  split(const std::string& str, const std::string& delims=" \t", char quote = '\'')
   {
     std::vector<std::string> res;
     size_t init = 0;
 
     while (true) {
-      auto pos = str.find_first_of(delims, init);
-      if (pos == std::string::npos) {
-        res.emplace_back(str.substr(init, str.length()));
-        break;
+      if (str[init] == quote) {
```

Comment:
How would you represent a quote in this case?

Reply:
What would you suggest as a test case?

Reply:
No longer relevant.